src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -275,47 +275,42 @@ public Set<Exercise> loadExercisesWithInformationForDashboard(Set<Long> exercise
     }
 
     /**
-     * Finds all Exercises for a given Course
+     * Filter all exercises for a given course based on the user role and course settings
+     * Assumes that the exercises are already been loaded (i.e. no proxy)
      *
-     * @param course corresponding course
+     * @param course corresponding course: exercises
      * @param user   the user entity
-     * @return a List of all Exercises for the given course
+     * @return a set of all Exercises for the given course
      */
-    public Set<Exercise> findAllForCourse(Course course, User user) {
-        Set<Exercise> exercises = null;
-        if (authCheckService.isAtLeastTeachingAssistantInCourse(course, user)) {
-            // tutors/instructors/admins can see all exercises of the course
-            exercises = exerciseRepository.findByCourseIdWithCategories(course.getId());
-        }
-        else if (authCheckService.isOnlyStudentInCourse(course, user)) {
+    public Set<Exercise> filterExercisesForCourse(Course course, User user) {
+        Set<Exercise> exercises = course.getExercises();
+        // no need to filter for tutors/editors/instructors/admins because they can see all exercises of the course
+        if (authCheckService.isOnlyStudentInCourse(course, user)) {
```

Comment:
I'd suggest exiting the method for better readability / less indentation levels:
```java
        if (authCheckService.isAtLeastTeachingAssistantInCourse(course, user)) {
            return;
        }
        // student code here
```

Reply:
Return does not work here, because further below there is still some logic independent of the if statement here.

The method is not very well designed and named. In addition to filtering, it also retrieves some additional data for team exercises and for currently active quiz exercises.

We should probably split it into two methods.